File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = Integer.MIN_VALUE;
        int cur = max;
        for (int i = 0; i < nums.length; i++){
            if (cur < 0){
                cur = nums[i];
            } else {
                cur += nums[i];
            }
            if (cur > max){
                max = cur;
            }
            // if 
        }


        return max;   
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int res = nums[0];
        int cur = nums[0];
        for (int i = 1; i < nums.Length; i++){
            cur = (cur + nums[i] > nums[i]) ? cur + nums[i] : nums[i];
            res = (cur > res) ? cur : res;
        }
        return res;
    }
}